stream_alert_cli/manage_lambda/rollback.py

```diff
@@ -13,128 +13,59 @@
-
-    Args:
-        lambda_config (dict): Lambda function config with 'current_version'
-
-    Returns:
-        True if the version was changed, False otherwise
-    """
-    current_version = lambda_config['current_version']
-    if current_version == '$LATEST':
-        return False
-
-    int_version = int(current_version)
-    if int_version <= 1:
-        return False
-
-    lambda_config['current_version'] = int_version - 1
-    return True
-
-
-def _try_decrement_version(lambda_config, function_name):
-    """Log a warning if the lambda version cannot be rolled back."""
-    changed = _decrement_version(lambda_config)
-    if not changed:
-        LOGGER_CLI.warn('%s cannot be rolled back from version %s',
-                        function_name, str(lambda_config['current_version']))
-    return changed
-
-
-def _rollback_alert(config):
-    """Decrement the current_version for the alert processor."""
-    lambda_config = config['lambda']['alert_processor_config']
-    if _try_decrement_version(lambda_config, 'alert_processor'):
-        return ['module.alert_processor_lambda']
-
-
-def _rollback_alert_merger(config):
-    """Decrement the current_version for the alert merger."""
-    lambda_config = config['lambda']['alert_merger_config']
-    if _try_decrement_version(lambda_config, 'alert_merger'):
-        return ['module.alert_merger_lambda']
-
-
-def _rollback_apps(config, clusters):
-    """Decrement the current_version for all of the apps functions in the given clusters."""
-    tf_targets = []
-
-    for cluster in clusters:
-        apps_config = config['clusters'][cluster]['modules'].get('stream_alert_apps', {})
-        for lambda_name, lambda_config in apps_config.iteritems():
-            if _try_decrement_version(lambda_config, lambda_name):
-                module_prefix = 'app_{}_{}'.format(lambda_config['app_name'], cluster)
-                tf_targets.append('module.{}'.format(module_prefix))
-
-    return tf_targets
-
-
-def _rollback_athena(config):
-    """Decrement the current_version for the Athena Partition Refresh function."""
-    lambda_config = config['lambda'].get('athena_partition_refresh_config')
-    if lambda_config and _try_decrement_version(lambda_config, 'athena_partition_refresh'):
-        return['module.stream_alert_athena']
-
-
-def _rollback_downloader(config):
-    """Decrement the current_version for the Threat Intel Downloader function."""
-    lambda_config = config['lambda'].get('threat_intel_downloader_config')
-    if lambda_config and _try_decrement_version(lambda_config, 'threat_intel_downloader_config'):
-        return['module.threat_intel_downloader']
 
+def _rollback_production(lambda_client, function_name):
+    """Rollback the production alias for the given function name."""
+    response = lambda_client.get_alias(FunctionName=function_name, Name='production')
+    current_version = int(response['FunctionVersion'])
+    if current_version <= 1:
```

Comment:
The version will never be less than 1

Reply:
And it's ok to roll back to `$LATEST`

Reply:
True, less than 1 is a silly check, I'll get rid of it (to avoid any confusion)

Because Terraform publishes new versions for every change, `$LATEST` will always be identical to the most recent published version. And if it were not (in case of a manual change), `$LATEST` will be, by definition, newer than any other version. So I would argue it's impossible to "roll back" to `$LATEST`